pkg/cloud/externalconn/impl_registry.go

```diff
@@ -16,23 +16,86 @@ import (
 	"net/url"
 
 	"github.com/cockroachdb/cockroach/pkg/base"
+	"github.com/cockroachdb/cockroach/pkg/cloud/externalconn/connectionpb"
 	"github.com/cockroachdb/cockroach/pkg/security/username"
 	"github.com/cockroachdb/errors"
 )
 
+type schemeRegistration struct {
+	FactoryType
+	connectionpb.ConnectionProvider
+	connectionParserFactory
+	validations map[string]ValidationFn
+}
+
 // parseAndValidateFns maps a URI scheme to a constructor of instances of that external
 // connection.
-var parseAndValidateFns = map[string]connectionParserFactory{}
+var parseAndValidateFns = map[string]schemeRegistration{}
+
+// FactoryType identifies the expected parsing logic used to create an external connection.
```

Comment:
nit: I prefer having all consts and type declarations at the top of the file, but I leave it up to you 🙂